src/functionalTest/java/uk/gov/hmcts/reform/jobscheduler/controllers/JobsControllerTest.java

```diff
@@ -26,31 +38,146 @@
-                    + "\"total_pages\":0,"
-                    + "\"total_elements\":0,"
-                    + "\"last\":true,"
-                    + "\"size\":10,"
-                    + "\"number\":0,"
-                    + "\"sort\":null,"
-                    + "\"first\":true,"
-                    + "\"number_of_elements\":0"
-                    + "}"
-            ))
+            .andExpect(jsonPath("content").isEmpty())
+            .andExpect(jsonPath("total_pages").value(0))
+            .andExpect(jsonPath("total_elements").value(0))
+            .andExpect(jsonPath("last").value(true))
+            .andExpect(jsonPath("size").value(10))
+            .andExpect(jsonPath("number").value(0))
+            .andExpect(jsonPath("sort").value(isNull()))
+            .andExpect(jsonPath("first").value(true))
+            .andExpect(jsonPath("number_of_elements").value(0));
+    }
+
+    @Test
+    public void should_create_a_new_job_and_return_job_uri_of_newly_created_job_when_valid_job_input_is_passed()
+        throws Exception {
+        //Given
+        String jobJson = jobJson();
+
+        //when
+        MvcResult mvcResult = sendPostRequest(jobJson)
+            .andExpect(status().isCreated())
+            .andReturn();
+
+        //then
+        //Create a new URI object which will validate whether the url we got back is an uri or not.
+        URI newJobUri = new URI(mvcResult.getResponse().getRedirectedUrl());
+
+        assertThat(newJobUri).isNotNull();
+
+        //Delete the newly created job else it will used by other test cases
+        sendDeleteRequest(extractJobId(mvcResult.getResponse().getRedirectedUrl()))
+            .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void should_throw_400_bad_request_when_empty_job_input_is_passed()
+        throws Exception {
+        sendPostRequest("")
+            .andExpect(status().isBadRequest())
+            .andReturn();
+    }
+
+    @Test
+    public void should_delete_job_successfully_when_existing_job_id_is_passed() throws Exception {
+        //Given
+        String jobJson = jobJson();
+
+        //when
+        MvcResult mvcResult = sendPostRequest(jobJson)
+            .andExpect(status().isCreated())
+            .andReturn();
+
+        //then
+        String jobId = extractJobId(mvcResult.getResponse().getRedirectedUrl());
+
+        sendDeleteRequest(jobId)
+            .andExpect(status().isNoContent());
+
+        //confirm the job is deleted
+        sendDeleteRequest(jobId)
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void should_throw_job_not_found_exception_when_non_existing_job_id_is_passed()
+        throws Exception {
+        MvcResult mvcResult = sendDeleteRequest(UUID.randomUUID().toString())
+            .andExpect(status().isNotFound())
             .andReturn();
+
+        assertThat(mvcResult.getResolvedException())
+            .isExactlyInstanceOf(JobNotFoundException.class);
```

Comment:
do we care about internal exceptions in functional tests?

Reply:
I had this test from perspective that if you receive a request to delete a job which does not exists then api will throw `JobNotFoundException`.

Reply:
imho here checking that it's 404 is enough. The exception should hopefully be checked in unit tests.

Reply:
ohhk now I got you 👍